x/precisebank/keeper/fractional_balance.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/precisebank/types"
+)
+
+// GetFractionalBalance returns the fractional balance for an address.
+func (k *Keeper) GetFractionalBalance(
+	ctx sdk.Context,
+	address sdk.AccAddress,
+) (sdkmath.Int, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.FractionalBalancePrefix)
+
+	bz := store.Get(types.FractionalBalanceKey(address))
+	if bz == nil {
+		return sdkmath.ZeroInt(), false
+	}
+
+	var bal sdkmath.Int
+	if err := bal.Unmarshal(bz); err != nil {
+		panic(fmt.Errorf("failed to unmarshal fractional balance: %w", err))
+	}
+
+	return bal, true
+}
+
+// SetFractionalBalance sets the fractional balance for an address.
+func (k *Keeper) SetFractionalBalance(
+	ctx sdk.Context,
+	address sdk.AccAddress,
+	amount sdkmath.Int,
+) {
+	if amount.IsZero() {
+		k.DeleteFractionalBalance(ctx, address)
+		return
+	}
+
+	// Ensure the fractional balance is valid before setting it. Use the
+	// NewFractionalAmountFromInt wrapper to use its Validate() method.
+	if err := types.NewFractionalAmountFromInt(amount).Validate(); err != nil {
```

Comment:
it might make more sense to enforce the caller to pass in the `FractionalAmount` and just call `Validate()` on that?

Reply:
hmm yeah I first tested with using `FractionalAmount` as the arg, but it seemed a bit strange since we'll only have `sdkmath.Int` after main logic and we'd have to wrap it every time its set.

I think an alternative is refactoring `NewFractionalAmountFromInt.Validate()` to just a regular function instead of creating a new wrapper type?

Created #1917 for discussion on refactor